src/NServiceBus.Core/BusConfiguration.cs

```diff
@@ -267,11 +267,18 @@ internal InitializableEndpoint Build()
         {
             ForAllTypes<T>(types, t =>
             {
+                if (!HasDefaultConstructor(t))
+                {
+                    throw new ArgumentException($"Unable to create the type '{t.Name}'. Types implementing '{typeof(T).Name}' must have a parameterless (default) constructor.");
```

Comment:
@SimonCropp got a better idea for the exception?

The overload checks for a public, instance constructor with no arguments, so a default ctor. If it doesn't find it, it should throw. Does `Activator.CreateInstance` work on types with an internal ctor?